gbdk-lib/examples/cross-platform/text_basic/src/main.c

```diff
@@ -0,0 +1,76 @@
+#include <gbdk/platform.h>
+#include <stdint.h>
+#include "Font.h"
+
+
+uint8_t GetCharacterVRamTile(char character) {
+
+    uint8_t vramTile=0;
+
+
+    // Char's can be interpreted as integers
+    // We don't need to map every alpha-numeric character
+    // We can use basic math to simplify A-Z and 0-9
+    if(character>='a'&&character<='z')vramTile = (character-'a')+1;
+    else if(character>='A'&&character<='Z')vramTile = (character-'A')+1;
+    else if(character>='0'&&character<='9')vramTile = (character-'0')+27;
+    else if(character=='!')vramTile = 37;
+    else if(character==':')vramTile = 38;
+    else if(character=='?')vramTile = 39;
+    else if(character=='/')vramTile = 40;
+    else if(character=='=')vramTile = 41;
+    else if(character==',')vramTile = 42;
+    else if(character=='.')vramTile = 43;
+    else if(character=='<')vramTile = 44;
+    else if(character=='>')vramTile = 45;
+
+
+    return vramTile;
+
+}
+
+void DrawText(uint8_t column, uint8_t row, char* text){
+
+    // Get the address of the first tile in the row
+    uint8_t* vramAddress = get_bkg_xy_addr(column,row);
+
+    uint16_t index=0;
+
+    while(text[index]!='\0'){
+
+        char character = text[index];
+
+        // Draw our character at the address
+        // THEN, increment the address
+        uint8_t vramTile = GetCharacterVRamTile(character);
+
+        set_vram_byte(vramAddress++,vramTile);
+
+        #if defined(SEGA)
+        set_vram_byte(vramAddress++,0);
+        #endif
+
+        index++;
+
+    }
+}
+
+void main(void)
+{
+    SHOW_BKG;
+
+    #if !defined(SEGA) && !defined(NINTENDO_NES)
```

Comment:
For the basic example which doesn't have sound effects it's better to keep it simple and omit the GB sound init code.